doc/development.rst

```diff
@@ -192,6 +192,56 @@ Upon a new release, the PEtab editors ensure that
 * the new version of the specifications is deposited at Zenodo
 * the new release is announced on the PEtab mailing list
 
+PEtab Extensions
+----------------
+
+An elaborate, monolithic format would make it difficult to understand and
+implement support for PEtab, leading to a steep learning curve and discouraging
+support in new toolboxes. To address this issue, the PEtab format is modular and
+permits modifications through extensions that complement the core standard.
+This modular specification evens the learning curve and provides toolbox
+developers with more guidance on which features to implement to maximize
+support for real world applications. Moreover, such modular extensions
+facilitate and promote the use of specialized tools for specific, non-parameter
+estimation tasks such as visualization.
+
+Requirements for new extensions:
+
+* Specifications in PEtab extensions take precedence over PEtab core, i.e., they
+  can ease or refine format restrictions imposed by PEtab core.
+* PEtab extensions should extend PEtab core with new orthogonal features or
+  tasks, i.e., they should not make trivial changes to PEtab core.
+* PEtab extensions must be named according to ^[a-zA-Z][\w\-]*$
+* PEtab extensions must be versioned using semantic versioning.
+* PEtab extensions required for interpretation of a problem specification must
+  be specified in the PEtab-YAML files
+* There is at least one tool that supports the proposed extension
+* The authors provide a library that provides test cases and implements
+  validation checks for the proposed format.
+
+Developers are free to develop any PEtab extension. To become an official
+PEtab extension, it needs to go through the following process.
+
+#. The developers write a proposal describing the motivation and specification
+   of the extension, following the respective issue template provided in this
+   repository.
+#. The proposal is submitted as an issue in this repository.
+#. The technical specification and documentation of the extension is submitted
+   as a pull request in this repository that references the respective issue.
+
+The PEtab editors jointly decide whether an extension meets the requirements
+described here. In case of a positive evaluation, they announce a poll for the
+acceptance as official extension to the PEtab forum. All members of the PEtab
+community are eligible to vote. If at least 50% of the votes are in favor,
+the extension is accepted and the respective pull requests with specifications,
+documentation and test cases are merged. There is no quorum number of votes
```

Comment:
Are there test cases for extensions here? This repo has no test cases, and earlier it is written that the extension should provided a library with test cases -- so that library would have already merged all PRs with test cases by now. A link to the library and test cases should be included in this repo though.

Reply:
> Are there test cases for extensions here?

Probably not.

I think it still needs to be sorted out whether all test cases should go to https://github.com/PEtab-dev/petab_test_suite or to an extension-specific repository. Also for the specs --- here or separate repo. I would leave this as is for now.